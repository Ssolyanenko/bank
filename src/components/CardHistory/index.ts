export * from './History';
export * from './Pagination';
export * from './CardHistory';
