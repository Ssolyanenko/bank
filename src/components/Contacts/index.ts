export * from './Contacts';
export * from './ContactsFooter';
export * from './ContactsInformation';
