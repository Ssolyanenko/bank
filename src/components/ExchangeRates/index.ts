export * from './ExchangeRates';
export * from './Converter';
export * from './Rates';
