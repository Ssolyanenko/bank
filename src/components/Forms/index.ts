export * from './Login';
export * from './PasswordRecoveryForm';
export * from './GeneralInfo';
export * from './Security';
export * from './OrderDebitTravelCard';
export * from './SetPinCodeForm';
export * from './OrderCreditCardPremium';
export * from './FormNotifications';
export * from './FilterHistoryForm';
export * from './CardLimitForm';
export * from './OrderCreditCardBilliable';
export * from './OrderDebitSmartCard';
export * from './TransferByPhoneNumberForm';
export * from './AcceptTerms';
