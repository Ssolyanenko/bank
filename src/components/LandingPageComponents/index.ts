export * from './LandingPageHeader';
export * from './LandingPageMain';
export * from './LandingPageFooter';
