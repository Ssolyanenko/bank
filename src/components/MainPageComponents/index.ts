export * from './FeaturedServices';
export * from './Menu';
export * from './DebitCards';
export * from './OperationHistory';
export * from './CreditCards';
export * from './CardProductsInner';
export * from './DebitSmartCard';
export * from './DebitTravelCard';
export * from './TransferList';
