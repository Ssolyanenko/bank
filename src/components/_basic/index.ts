export * from './PrimaryButton';
export * from './Button';
export * from './Input';
export * from './InputPassword';
export * from './ModalWrapper';
export * from './ModalConfirmation';
export * from './PhoneCodeTimer';
export * from './ModalBox';
export * from './CloseButton';
export * from './BackButton';
export * from './StepperForm';
export * from './StepperLabel';
export * from './ChatButton';
export * from './BanTimer';
export * from './CardTemplate';
export * from './SecondaryButton';
export * from './generic';
export * from './SelectStyled';
export * from './TextFieldStyled';
export * from './InputStyled';
export * from './CardTemplate';
export * from './Modal';
export * from './SecureWrapper';
export * from './InputRange';
export * from './DropDown';
export * from './RadioButtonsGroup';
export * from './FormError';
export * from './Checkbox';
export * from './Switcher';
export * from './EditButton';
export * from './CardBlock';
export * from './Tabs';
export * from './ModalSuccess';
export * from './DropDownWithCheckbox';
export * from './SelectFormikTemplate';
export * from './Loader';
export * from './DatePicker';
export * from './CustomRoutes';
