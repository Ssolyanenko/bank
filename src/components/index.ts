export * from './LandingPageComponents';
export * from './Forms';
export * from './ArrowsBlock';
export * from './RootModal';
export * from './Slider';
export * from './Map';
export * from './Contacts';
export * from './MainPageComponents';
export * from './ExchangeRates';
export * from './CardProducts';
export * from './CardHistory';
export * from './SetPinSuccessModal';
export * from './MyCardsApplications';
export * from './MyLoans';
export * from './MyCardsApplications';
export * from './InformationBlock';
export * from './StatusTag';
export * from './DropDownWithSearchBranches';
export * from './FormInputBox';
export * from './MyCards';
export * from './ActivateBlockCardModal';
export * from './UserAccount';
export * from './SetPinCodeFormModal';
export * from './CardDetails';
export * from './CardStatement';
export * from './CardStatementButtons';
export * from './Accordion';
export * from './ReceiptButtons';
export * from './TemporaryComponent/TemporaryComponent';
export * from './TransferModal';
export * from './ModalChoosePaymentCard';
export * from './CardInfo';
export * from './ShareReceiptModal/ShareReceiptModal';
export * from './SocialMedia';
export * from './ResetFormsConfirmModal';
export * from './TransactionLimit';
export * from './ChangeTransactionLimit';
export * from './ActivateBlockCard';
export * from './CardManage';
export * from './ChangePINCode';
export * from './ResponseModal';
export * from './TransactionLimit';
