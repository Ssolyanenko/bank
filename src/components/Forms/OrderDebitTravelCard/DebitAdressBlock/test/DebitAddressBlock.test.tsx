import 'jsdom-global/register';
import React, { ReactElement } from 'react';
import { mount, shallow } from 'enzyme';
import { Formik } from 'formik';
import { act } from 'react-dom/test-utils';

import { DebitAddressBlock } from 'components';
import { ERROR_MESSAGE } from 'constants/errors';

jest.mock('react-i18next', () => ({
  useTranslation: () => ({
    t: (str: string) => str,
    i18n: {
      changeLanguage: () => new Promise(jest.fn()),
    },
  }),
  initReactI18next: {
    type: '3rdParty',
    init: jest.fn(),
  },
}));

describe('<DebitAddressBlock />', (): void => {
  test('should make a snapshot of the component', (): void => {
    const wrapper = shallow(
      <Formik
        initialValues={{ city: '', house: '', street: '', unit: '', apartment: '', postCode: '' }}
        onSubmit={jest.fn()}
      >
        {({ errors, touched }): ReactElement => (
          <DebitAddressBlock
            setFieldValue={jest.fn()}
            setFieldTouched={jest.fn()}
            isSendButtonClicked={false}
            errors={errors}
            touched={touched}
          />
        )}
      </Formik>
    );
    const component = wrapper.getElement();

    expect(component).toMatchSnapshot();
  });

  test('should call Delivery Address useEffect change', (): void => {
    const useEffectMock = jest.spyOn(React, 'useEffect').mockImplementation((f) => f());

    const setFieldValue = jest.fn();
    const setFieldTouched = jest.fn();

    const wrapper = mount(
      <Formik
        initialValues={{ city: '', house: '', street: '', unit: '', apartment: '', postCode: '' }}
        onSubmit={jest.fn()}
      >
        {({ errors, touched }): ReactElement => (
          <DebitAddressBlock
            setFieldValue={setFieldValue}
            setFieldTouched={setFieldTouched}
            isSendButtonClicked={false}
            errors={errors}
            touched={touched}
          />
        )}
      </Formik>
    );

    wrapper.find('input[name="street"]').simulate('change', { target: { name: 'street', value: 'London' } });
    wrapper.update();
    expect(useEffectMock).toHaveBeenCalled();
  });

  test('should show errors', (): void => {
    const wrapper = mount(
      <Formik
        initialValues={{ city: '', house: '', street: '', unit: '', apartment: '', postCode: '' }}
        onSubmit={jest.fn()}
      >
        {({ errors, touched }): ReactElement => (
          <DebitAddressBlock
            setFieldValue={jest.fn()}
            setFieldTouched={jest.fn()}
            isSendButtonClicked={true}
            errors={{
              ...errors,
              street: ERROR_MESSAGE.required,
              city: ERROR_MESSAGE.required,
              postCode: ERROR_MESSAGE.required,
              house: ERROR_MESSAGE.required,
            }}
            touched={touched}
          />
        )}
      </Formik>
    );

    expect(wrapper.find('.error').length).toBe(4);
  });
});
