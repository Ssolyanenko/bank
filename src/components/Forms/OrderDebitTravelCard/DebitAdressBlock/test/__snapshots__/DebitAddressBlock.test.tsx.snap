// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DebitAddressBlock /> should make a snapshot of the component 1`] = `
<Context.Provider
  value={
    {
      "dirty": false,
      "errors": {},
      "getFieldHelpers": [Function],
      "getFieldMeta": [Function],
      "getFieldProps": [Function],
      "handleBlur": [Function],
      "handleChange": [Function],
      "handleReset": [Function],
      "handleSubmit": [Function],
      "initialErrors": {},
      "initialStatus": undefined,
      "initialTouched": {},
      "initialValues": {
        "apartment": "",
        "city": "",
        "house": "",
        "postCode": "",
        "street": "",
        "unit": "",
      },
      "isSubmitting": false,
      "isValid": true,
      "isValidating": false,
      "registerField": [Function],
      "resetForm": [Function],
      "setErrors": [Function],
      "setFieldError": [Function],
      "setFieldTouched": [Function],
      "setFieldValue": [Function],
      "setFormikState": [Function],
      "setStatus": [Function],
      "setSubmitting": [Function],
      "setTouched": [Function],
      "setValues": [Function],
      "status": undefined,
      "submitCount": 0,
      "submitForm": [Function],
      "touched": {},
      "unregisterField": [Function],
      "validateField": [Function],
      "validateForm": [Function],
      "validateOnBlur": true,
      "validateOnChange": true,
      "validateOnMount": false,
      "values": {
        "apartment": "",
        "city": "",
        "house": "",
        "postCode": "",
        "street": "",
        "unit": "",
      },
    }
  }
>
  <DebitAddressBlock
    errors={{}}
    isSendButtonClicked={false}
    setFieldTouched={[MockFunction]}
    setFieldValue={[MockFunction]}
    touched={{}}
  />
</Context.Provider>
`;
